components/engine_traits/src/cf_options.rs

```diff
@@ -0,0 +1,10 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+pub trait CFOptions {
+    fn get_level_zero_slowdown_writes_trigger(&self) -> u32;
+    fn get_level_zero_stop_writes_trigger(&self) -> u32;
+    fn get_soft_pending_compaction_bytes_limit(&self) -> u64;
+    fn get_hard_pending_compaction_bytes_limit(&self) -> u64;
+    fn get_block_cache_capacity(&self) -> u64;
+    fn set_block_cache_capacity(&self, capacity: u64) -> Result<(), String>;
```

Comment:
I don't think these options should be a common part of engines. It is highly coupled with RocksDB implementation.

Reply:
Perhaps in the long term, yes. For now I think the best way forward is to 'firewall' off the concrete rocks engine behind _some_ abstraction so that other engines can proceed. Then those abstractions that look too much like rocks can be themselves refactored as the needs of other engines arise.

One thing at a time.